csrc/keyutils.h

```diff
@@ -12,83 +12,63 @@
 #include <openssl/x509.h>
 #include "util.h"
 #include "env.h"
+#include "auto_free.h"
 
 namespace AmazonCorrettoCryptoProvider {
 
+// This class should generally not be used for new development
+// as it has been replaced by the *_auto classes in auto_free.h
+// The only time this class should be used is when you *need* to keep various EVP objects together.
+// The only currently known *good* use for this class is tracking state when signing/verifying data.
```

Comment:
is that something we'd want to (eventually) create special-purpose `auto_`-wrapped buffer for?

Reply:
Doesn't seem likely. That wrapper is to make using things from libcrypto easier. This bundles a few of them together so they live together.